spark-cassandra-connector/src/main/scala/org/apache/spark/sql/cassandra/ConfCache.scala

```diff
@@ -0,0 +1,133 @@
+package org.apache.spark.sql.cassandra
+
+import com.datastax.spark.connector.cql.CassandraConnectorConf
+import com.datastax.spark.connector.rdd.ReadConf
+import com.datastax.spark.connector.writer.WriteConf
+import org.apache.commons.lang.StringUtils
+
+import scala.collection.mutable
+
+
+object ConfCache {
```

Comment:
I'm still not convinced. I know that singletons simplifies things to some degree but at the end they may show some problems. I know that you know that. Why can't we get the settings from `SQLConf`?  The user may want to create two separate SQL contexts and provide different configurations for them, for example because one of the will be used with streaming and the other to analyse the collected data.